### ISSUE TITLE
Projects in a Turborepo monorepo not using workspace package.json config

### ISSUE THREAD

**Issue by jbadavis** [xojs/xo#733]
Hello,

I have a very simple XO configuration I'd like to share between all the packages and apps in my Turborepo monorepo:

```json
"xo": {
  "prettier": true,
  "rules": {
    "import/extensions": 0
  }
}
```

The project README states that I should just need one version of this in the root package.json file:

- https://github.com/xojs/xo#monorepo

However, I'm seeing linting errors in each of my packages if I don't have this XO config in every package.json. Have I configured something incorrectly?

**Comment by Navigatron** [comment]
I'm seeing the same behavior / having the same problem. Running `xo` in a sub-package does not seem to "search upwards" and find the parent package.json config.

```
- package.json         # Includes XO config
- packages/
    - packageName/
        - package.json # Does not include XO config
```

I believe there are some magic filenames that might work - I'll try those out and report back. I'll watch this issue with interest though, as I'd rather not have yet another root config file.

**Comment by Navigatron** [comment]
I've found the discrepancy!

From lines [103](https://github.com/xojs/xo/blob/37e14580c275a15fa36435d0cda7d45d3c515828/lib/options-manager.js#L103) and [104](https://github.com/xojs/xo/blob/37e14580c275a15fa36435d0cda7d45d3c515828/lib/options-manager.js#L104C3-L104C3) of `lib/options-manager.js`:

```js
options.cwd = path.resolve(options.cwd || process.cwd());
const configExplorer = cosmiconfig(MODULE_NAME, {searchPlaces: CONFIG_FILES, loaders: {noExt: defaultLoaders['.json']}, stopDir: options.cwd});
```

Note that `stopDir` is set to `options.cwd`, which defaults to `process.cwd()`.

From the [comsiconfig readme](https://github.com/cosmiconfig/cosmiconfig/blob/42ca3fab6d6ae593a895c0fe4e2a5f6b297e6361/README.md?plain=1#L154C11-L154C11):

> - If none of those searches reveal a configuration object, move up one directory level and try again. So the search continues in ./, ../, ../../, ../../../, etc., checking the same places in each directory.
> - Continue searching until arriving at your home directory (or some other directory defined by the cosmiconfig option [stopDir](https://github.com/cosmiconfig/cosmiconfig#stopdir)).

In short, xo does not search upwards - the search is stopped in the current working directory.

# Workaround

1. Use the `--cwd` flag to start xo from your monorepo root (overriding options.cwd)
2. Narrow scope back down by specifying the subfolder you want

```json
{
    "scripts": {
        "lint": "xo --cwd=../.. $(pwd)"
    }
}
```
 There's probably a better way to identify the monorepo root than just hard-coding `../..`, but that's all I've got for now.

**Comment by jamestalmage** [comment]
I'm having issues even if I ignore the recommendation and do a config per child package. All configs are ignored if run from the child cwd
